data/templates/esx-ks

```diff
@@ -79,7 +79,12 @@ esxcli system settings advanced set -o /UserVars/ESXiShellInteractiveTimeout -i
 esxcli network firewall set --default-action false --enabled no
 
 # set hostname and domain
-esxcli system hostname set --domain <%=domain%> --host <%=hostname%>
+# if hostname is 'localhost', domain cannot be set.
+<% if (domain) { %>
```

Comment:
There is a special case that if the `domain` is not specified rather than `null`, this rendering will fail as it cannot find the reference of `domain`. If the Task-Schema takes effect, the domain cannot be set to null, so your code will fail. I suggest changing to following syntax: 
https://github.com/RackHD/on-http/blob/master/data/templates/centos-ks#L83

Meanwhile, you still need to delete the `domain` from options list if it is null, just like the `kvm`:
https://github.com/RackHD/on-tasks/blob/master/lib/jobs/install-os.js#L251-L253